### PR TITLE
add macos project to macrobenchmarks

### PR DESCRIPTION
**Pull request by dnfield**
Adds a macos project to macrobenchmarks for local testing. We do not yet have any benchmarks for this.

/cc @liyuqian @stuartmorgan

### CODE REVIEW

**Comment by liyuqian** [comment]
I wonder why the original PR (https://github.com/flutter/flutter/pull/50851) has `+1,317` lines while this sub-PR has `+17,830` lines. Is everything in `ephemeral` necessary?

**Comment by dnfield** [comment]
Ahh good catch, I'll fix that.  I added `macos` and forgot to update the .gitignore.

**Comment by dnfield** [comment]
I think this is right now.

**Review by liyuqian — APPROVED** [review]
RSLGTM. Defer the final LGTM to the desktop experts.

File: dev/benchmarks/macrobenchmarks/macos/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (stuartmorgan) [inline dev/benchmarks/macrobenchmarks/macos/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata]: This file is a mistake that I didn't notice was added to the template. Someone just caught this and sent a PR recently that fixed the template, but this file should be removed here.

File: dev/benchmarks/macrobenchmarks/macos/Runner/Configs/AppInfo.xcconfig [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+//
+// This may be replaced with something auto-generated from metadata (e.g., pubspec.yaml) in the
+// future. If not, the values below would default to using the project name when this becomes a
+// 'flutter create' template.
+
+// The application's name. By default this is also the title of the Flutter window.
+PRODUCT_NAME = macrobenchmarks
+
+// The application's bundle identifier
+PRODUCT_BUNDLE_IDENTIFIER = com.example.macrobenchmarks
```
Comment (stuartmorgan) [inline dev/benchmarks/macrobenchmarks/macos/Runner/Configs/AppInfo.xcconfig]: Do we use junk bundle ID and copyright in the iOS app? If not, these should be fixed up.

**Review by stuartmorgan-g — APPROVED** [review]
LGTM with two small things

File: dev/benchmarks/macrobenchmarks/macos/Runner/Configs/AppInfo.xcconfig (hunk 2 above)
Comment (dnfield) [reply dev/benchmarks/macrobenchmarks/macos/Runner/Configs/AppInfo.xcconfig]: Yes - these values match the iOS versions.

File: dev/benchmarks/macrobenchmarks/macos/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata (hunk 1 above)
Comment (dnfield) [reply dev/benchmarks/macrobenchmarks/macos/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata]: Removed

**Pull request merged**